### PR TITLE
Add basic regex validation for common types

### PR DESCRIPTION
**Pull request by fitore-cds**
# Summary | Résumé

This PR adds the ability to specify the type of the field (email, number, text, phone, date) or a custom regex in the `validation` object in the Form config.

Currently, only the Platform Intake form takes these extra validators under account, but we should introduce them to our demos moving forward.

@bryan-robitaille I'm debating whether to introduce [Yup](https://github.com/jquense/yup) now or as a later iteration?

### CODE REVIEW

File: lib/validation.tsx [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
-import { FormikProps } from "formik";
+import {
+  FormValues,
+  DynamicFormProps,
+  InnerFormProps,
+  FormElement,
+  Responses,
+  ValidationProperties,
+} from "./types";
+import { FormikProps, FormikValues } from "formik";
```
Comment (bryan-robitaille) [inline lib/validation.tsx:10]: Just a quick removal of `FormikValues` which isn't being used in this file.

File: lib/validation.tsx [hunk 2]
```diff
@@ -52,3 +69,41 @@ export const getErrorList = (
...
+    };
+  }
+
+  const REGEX_CONFIG: RegexProps = {
+    email: {
+      regex: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
+      error: t("input-validation.email"),
+    },
+    text: {
+      regex: /^[0-9a-zA-Z]+$/,
```
Comment (bryan-robitaille) [inline lib/validation.tsx:87]: Maybe we should add`-` as a valid character given hyphenated composed names?

File: lib/validation.tsx [hunk 3]
```diff
@@ -52,3 +69,41 @@ export const getErrorList = (
...
+      error: t("input-validation.text"),
+    },
+    number: {
+      regex: /^[0-9]+$/,
```
Comment (bryan-robitaille) [inline lib/validation.tsx:91]: Maybe we should add `,` and `.` as valid characters as well given some people use `,` to separate every 10<sup>3</sup> and `.` due to floats.

**Review by bryan-robitaille — CHANGES_REQUESTED** [review]
I think some quick changes to the regex's would really benefit the UX for now until we have more information coming back from research.

File: lib/validation.tsx (hunk 1 above)
Comment (fitore-cds) [reply lib/validation.tsx]: Good catch, thank you!

File: lib/validation.tsx (hunk 3 above)
Comment (fitore-cds) [reply lib/validation.tsx]: Yes, good idea I updated the regex for text and number.

**Review by bryan-robitaille — APPROVED** [review]
Awesome!

**Pull request merged**